homeassistant/components/meater/sensor.py

```diff
@@ -0,0 +1,175 @@
+from homeassistant.helpers.update_coordinator import (
+    CoordinatorEntity,
+    DataUpdateCoordinator,
+    UpdateFailed,
+)
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup(hass: HomeAssistant, config: dict):
+    """Set up the Meater Temperature Probe sensor."""
+    return True
+
+
+async def async_setup_entry(hass, entry, async_add_entities):
+    """Set up the entry."""
+    # assuming API object stored here by __init__.py
+    api = hass.data[DOMAIN][entry.entry_id]
+
+    async def async_update_data():
+        """Fetch data from API endpoint."""
+        try:
+            # Note: asyncio.TimeoutError and aiohttp.ClientError are already
+            # handled by the data update coordinator.
+            async with async_timeout.timeout(10):
+                devices = await api.get_all_devices()
+        except AuthenticationError as err:
+            raise UpdateFailed("The API call wasn't authenticated") from err
+        except TooManyRequestsError as err:
+            raise UpdateFailed(
+                "Too many requests have been made to the API, rate limiting is in place"
+            ) from err
+        except Exception as err:  # pylint: disable=broad-except
+            raise UpdateFailed(f"Error communicating with API: {err}") from err
+
+        # Populate the entities
+        entities = []
+
+        for dev in devices:
+            if dev.id not in hass.data[DOMAIN]["entities"]:
+                entities.append(
+                    MeaterProbeTemperature(
+                        coordinator, dev.id, TemperatureMeasurement.Internal
+                    )
+                )
+                entities.append(
+                    MeaterProbeTemperature(
+                        coordinator, dev.id, TemperatureMeasurement.Ambient
+                    )
+                )
+                device_registry = await dr.async_get_registry(hass)
+                device_registry.async_get_or_create(
+                    config_entry_id=entry.entry_id,
+                    identifiers={(DOMAIN, dev.id)},
+                    name=f"Meater Probe {dev.id}",
+                    manufacturer="Apption Labs",
+                    model="Meater Probe",
+                )
+                hass.data[DOMAIN]["entities"][dev.id] = None
+
+        async_add_entities(entities)
+
+        return devices
+
+    coordinator = DataUpdateCoordinator(
+        hass,
+        _LOGGER,
+        # Name of the data. For logging purposes.
+        name="meater_api",
+        update_method=async_update_data,
+        # Polling interval. Will only be polled if there are subscribers.
+        update_interval=timedelta(seconds=30),
+    )
+
+    def null_callback():
+        return
+
+    # Add a subscriber to the coordinator that doesn't actually do anything, just so that it still updates when all probes are switched off
+    coordinator.async_add_listener(null_callback)
```

Comment:
I'm not following this one?

Reply:
If the config entry is setup with no temperature probes yet connected, there are no subscribers to the coordinator so we won't discover new probes. By adding the dummy subscriber we'll keep refreshing when there are no entities.